pages/setuppwa/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { useRouter } from "next/router"
+import React from "react"
+
+const SetupPwa = () => {
+  const router = useRouter()
+  const [username, setUsername] = React.useState<string>("")
+
+  const username_from_local = localStorage.getItem("username")
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+
+    if (!username_from_local) {
+      localStorage.setItem("username", username)
+    }
+
+    router.push(
+      {
+        pathname: `/merchant/${username}`,
+      },
+      undefined,
+      { shallow: true },
+    )
+  }
+
+  React.useEffect(() => {
+    if (username_from_local) {
+      router.push(
+        {
+          pathname: `/merchant/${username_from_local}`,
+        },
+        undefined,
+        { shallow: true },
+      )
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [username_from_local])
+
+  if (!username_from_local) {
+    return (
+      <div className="setup-pwa">
+        <form
+          className="username-form"
+          autoComplete="off"
+          onSubmit={(event: React.FormEvent<HTMLFormElement>) => handleSubmit(event)}
+        >
+          <h4>Welcome to Bitcoin beach POS application.</h4>
+          <label htmlFor="username">
+            To use the app, enter your bitcoin beach username
+          </label>
+          <input
+            type="text"
+            name="username"
+            value={username}
+            onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
+              setUsername(event.target.value)
+            }
+            placeholder="username"
+            required
+          />
+          <button>Submit</button>
+        </form>
+      </div>
+    )
+  }
+  return (
```

Comment:
You may not want to show a loader because it will just flash. It could be better to return null or a blank screen.

Reply:
It doesn't. It shoes the loading spinner until the page loads. I tested this on android and ios and it's the same pattern

Reply:
Doesn't it load almost instantaneously?

Reply:
Also hasn't the page already loaded if you are showing that? What is it waiting on to load?

Reply:
No. Only on PC does it loads instantly. Not sure why though.

Reply:
My guess is that's the behaviour of PWAs since they have to fetch the saved manifest and load the corresponding code before rendering the page

Reply:
> Also hasn't the page already loaded if you are showing that? What is it waiting on to load?

It shows the loader until one of the conditions in the if check is fulfilled.

Reply:
I wonder then if then if the loader is actually coming from another higher up component and not this page. It seems that if the page is rendering the loader then the page has already loaded. For curiosity sake can you see if the loader still shows up if you return null there?

Reply:
> I wonder then if then if the loader is actually coming from another higher up component and not this page. It seems that if the page is rendering the loader then the page has already loaded. For curiosity sake can you see if the loader still shows up if you return null there?

With null returned, It returns an empty page till the page loads. 
And Yes, the page is loaded with the default loader until a condition is fulfilled. The loader is definitely part of the component not else where. You can see it in line 66

Reply:
Sounds good. Thanks for checking!